### ISSUE TITLE
Webdriveragent build failed as no account found

### ISSUE THREAD

**Issue by rajeshj93** [Jitu1888/ios-device-lab#3]
WDA is failing because of error: No Accounts: Add a new account in Accounts settings. (in target 'WebDriverAgentRunner' from project 'WebDriverAgent')

When i trigger the WDA from xcode build is getting succeeded. Issue is only when i run MAKE command throught terminal. 

Could you please here ? @Jitu1888

**Comment by rajeshj93** [comment]
@Jitu1888 how did you overcome this error ? Can you please assist...

**Comment by rajeshj93** [comment]
@Jitu1888   can you please assist on this issue ??

**Comment by NguyenHoangDuy2406** [comment]
@rajeshj93 
Did you do all the below steps?

```
Add your developer Apple ID to XCode

    XCode -> XCode menu -> Preferences -> Accounts Tab
    Click + under Apple IDs list
    Choose Apple ID
    Login to your account

Download a "Apple Development certificate" for your user

    Continue from previous step, right after logging into your Developer account in Xcode
    Select Manage Certificates
    Click + in the lower left corner
    Select Apple Development

Clone the various needed repos ( includes WebDriverAgent )

    Run make clone

Configure WebDriverAgent to use your identity for signing

    Open repos/WebDriverAgent/WebDriverAgent.xcodeproj in XCode
    Select the WebDriverAgentLib target
    Go to the Signing & Capabilities tab
    Select your team under Team
    Select the WebDriverAgentRunner target
    Go to the Signing & Capabilities tab
    Select your team under Team
```

**Comment by rajeshj93** [comment]
@NguyenHoangDuy2406  
Yes i have followed same steps but when am trying to run make command it throws the above error. Whereas when i am trying to build the WDA through xcode it works.

**Comment by rajeshj93** [comment]
@NguyenHoangDuy2406 can you please assist here ?

**Comment by NguyenHoangDuy2406** [comment]
@rajeshj93 it's so strange :(. It's working OK my my side

**Comment by rajeshj93** [comment]
@NguyenHoangDuy2406 did we added our team name in make file as well ?

There we have a block of lines right with respect to build

wda: bin/wda/build_info.json

xcodebuildoptions1 := \
	-scheme WebDriverAgentRunner \
	-allowProvisioningUpdates \
	-destination generic/platform=iOS

DEVID := $(shell jq .xcode_dev_team_id config.json -j)

xcodebuildoptions2 := \
	CODE_SIGN_IDENTITY="iPhone Developer" \
	DEVELOPMENT_TEAM="$(DEVID)"

**Comment by rajeshj93** [comment]
@NguyenHoangDuy2406 can you please check this ?

**Comment by NguyenHoangDuy2406** [comment]
@rajeshj93 which MacOS model that you're using? Apple Silicon or Intel? Because in my experience, when I try `make` with Apple Silicon, it shouldn't work before. So now I'm using the Intel chip one

**Comment by rajeshj93** [comment]
I think i am using Intel only @NguyenHoangDuy2406

**Comment by Jitu1888** [comment]
@rajeshj93 better use https://github.com/AppiumTestDistribution/appium-device-farm. This also runs an appium session for iOS you will get access to UI also.

**Comment by Jitu1888** [comment]
I have tested it and it working fine now @rajeshj93

**Comment by Jitu1888** [comment]
@rajeshj93 You need to use an intel processor. Let me try an m2 chip, give me 2 days. I was busy with some tasks. Now i have time to fix all the issues

**Comment by rajeshj93** [comment]
> @rajeshj93 better use https://github.com/AppiumTestDistribution/appium-device-farm. This also runs an appium session for iOS you will get access to UI also.

@Jitu1888 so you want me to run the appiun device farm for ios in another terminal and when the grip is up and running you want me to try running make command so it will fetch the session and so the ios UI in the STF ? Please confirm...

**Comment by Jitu1888** [comment]
it's a different entity, on mac run the appium with plugins:
appium plugin install --source=npm appium-device-farm
appium plugin install --source=npm appium-dashboard
appium plugin install --source=npm appium-gestures-plugin
appium plugin install --source=npm appium-wait-plugin
appium plugin install --source=npm @appium/execute-driver-plugin

Install npm install -g go-ios
Run Appium with the below command:

appium server -ka 800 --use-plugins=device-farm --config ./server-config.json -pa /wd/hub

In dashboard you can see iOS and Android phones. You can control it. @rajeshj93 

server-config.json
{
    "server": {
        "use-plugins": [
            "device-farm",
            "appium-dashboard",
            "gestures",
            "execute-driver"
        ],
        "plugin": {
            "device-farm": {
                "skipChromeDownload": false,
                "platform": "android"
            }
        },
        "port": 4723
    }
}

Please follow the below URL 
https://devicefarm.org/setup/#installation-client
